lua/orgmode/org/hyperlinks.lua

```diff
@@ -1,61 +1,27 @@
 local Files = require('orgmode.parser.files')
 local utils = require('orgmode.utils')
+local fs = require('orgmode.utils.fs')
 local Hyperlinks = {}
 
-local function get_file_from_context(ctx)
-  local filepath = (ctx.hyperlinks and ctx.hyperlinks.filepath)
-  if not filepath then
-    return Files.get_current_file()
-  end
-  local canonical = vim.loop.fs_realpath(filepath)
-  if not canonical then
-    return Files.get_current_file()
-  end
-
-  return Files.get(canonical)
-end
-
-local function update_hyperlink_ctx(ctx)
-  if not ctx.line then
-    return
-  end
-
-  -- TODO: Support text search, see here [https://orgmode.org/manual/External-Links.html]
-  local hyperlinks_ctx = {
-    filepath = false,
-    headline = false,
-    custom_id = false,
-  }
-
-  local file_match = ctx.line:match('file:(.-)::')
-  if file_match then
-    file_match = Hyperlinks.get_file_real_path(file_match)
-  end
-
-  if file_match then
-    file_match = vim.loop.fs_realpath(file_match)
-  end
-
-  if file_match and Files.get(file_match) then
-    hyperlinks_ctx.filepath = Files.get(file_match).filename
-    hyperlinks_ctx.headline = ctx.line:match('file:.-::(%*.-)$')
-
-    if not hyperlinks_ctx.headline then
-      hyperlinks_ctx.custom_id = ctx.line:match('file:.-::(#.-)$')
-    end
-
-    ctx.base = hyperlinks_ctx.headline or hyperlinks_ctx.custom_id or ctx.base
-  end
-
-  ctx.hyperlinks = hyperlinks_ctx
+---@param url Url
+local function get_file_from_url(url)
+  local file_path = url:get_filepath()
+  local canonical_path = file_path and fs.get_real_path(file_path)
+  return canonical_path and Files.get(canonical_path) or Files.get_current_file()
 end
 
-function Hyperlinks.find_by_filepath(ctx)
+---@param url Url
+---@return string[]
+function Hyperlinks.find_by_filepath(url)
   local filenames = Files.filenames()
-  local file_base = ctx.base:gsub('^file:', '')
+  local file_base = url:get_filepath()
+  if not file_base then
+    return {}
+  end
+  --TODO integrate with orgmode.utils.fs or orgmode.objects.url
```

Comment:
I'm not sure, if we need this function anymore. I have to admit, that I didn't completely understand, which use-cases it actually tries to cover. It seems to only find org-files, which are already parsed. My gut-feeling tells me, that we could simplify it like in `find_by_custom_id` and `find_by_title`, but I wasn't sure, if I would remove something. @kristijanhusak can you enlighten me?

Reply:
The idea is to show all available org files in autocomplete popup when you start typing. Basically, typing this:
```
[[file:./
```
Should give you list of all org files that are part of `org_agenda_files` in your setup. We need to leave it, since  other functions will not return that.

Reply:
Thanks for the clarification.